controllers/neutronsriovagent_controller.go

```diff
@@ -1,163 +1,116 @@
-	err = c.Watch(&source.Kind{Type: &corev1.Secret{}}, &handler.EnqueueRequestForOwner{
-		IsController: false,
-		OwnerType:    &neutronv1.NeutronSriovAgent{},
-	})
-	if err != nil {
-		return err
-	}
-
-	// TODO(user): Modify this to be the types you create that are owned by the primary resource
-	// Watch for changes to secondary resource Pods and requeue the owner NeutronSriovAgent
-	err = c.Watch(&source.Kind{Type: &corev1.Pod{}}, &handler.EnqueueRequestForOwner{
-		IsController: true,
-		OwnerType:    &neutronv1.NeutronSriovAgent{},
-	})
-	if err != nil {
-		return err
-	}
-
-	return nil
-}
-
-// blank assignment to verify that ReconcileNeutronSriovAgent implements reconcile.Reconciler
-var _ reconcile.Reconciler = &ReconcileNeutronSriovAgent{}
+var ospHostAliases = []corev1.HostAlias{}
 
-// ReconcileNeutronSriovAgent reconciles a NeutronSriovAgent object
-type ReconcileNeutronSriovAgent struct {
-	// This client, initialized using mgr.Client() above, is a split client
-	// that reads objects from the cache and writes to the apiserver
-	client client.Client
-	scheme *runtime.Scheme
+// NeutronSriovAgentReconciler reconciles a NeutronSriovAgent object
+type NeutronSriovAgentReconciler struct {
+	Client client.Client
+	Log    logr.Logger
+	Scheme *runtime.Scheme
 }
 
-// Reconcile reads that state of the cluster for a NeutronSriovAgent object and makes changes based on the state read
-// and what is in the NeutronSriovAgent.Spec
-// TODO(user): Modify this Reconcile function to implement your Controller logic.  This example creates
-// a Pod as an example
-// Note:
-// The Controller will requeue the Request to be processed again if the returned error is non-nil or
-// Result.Requeue is true, otherwise upon completion it will remove the work from the queue.
-func (r *ReconcileNeutronSriovAgent) Reconcile(request reconcile.Request) (reconcile.Result, error) {
-	reqLogger := log.WithValues("Request.Namespace", request.Namespace, "Request.Name", request.Name)
+// Reconcile reconcile keystone API requests
+// +kubebuilder:rbac:groups=neutron.openstack.org,resources=neutronsriovagents,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups=neutron.openstack.org,resources=neutronsriovagents/status,verbs=get;update;patch
+// +kubebuilder:rbac:groups=core,resources=secrets,verbs=get;list;create;update;delete;
+// +kubebuilder:rbac:groups=core,resources=configmaps,verbs=get;list;create;update;delete;
+// +kubebuilder:rbac:groups=core,resources=services,verbs=get;list;create;update;delete;
+// +kubebuilder:rbac:groups=batch,resources=jobs,verbs=get;list;create;update;delete;
+// +kubebuilder:rbac:groups=apps,resources=deployments,verbs=get;list;create;update;delete;
+func (r *NeutronSriovAgentReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
+	_ = context.Background()
+	reqLogger := r.Log.WithValues("neutronsriovagent", req.NamespacedName)
 	reqLogger.Info("Reconciling NeutronSriovAgent")
```

Comment:
we could also just replace all `reqLogger.Info()` with `r.Log.Info()` for the log statements and keep the default `_ = r.Log.WithValues("neutronsriovagent", req.NamespacedName)`, but this is ok